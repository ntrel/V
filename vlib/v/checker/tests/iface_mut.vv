interface Register {
	//mut:
	register(int)
}

struct S1 {
mut:
	i int
}

fn (mut f S1) register(i int) {
	f.i = i
}

fn do(mut r Register) {
	r.register(8)
}

fn test_register() {
	mut t := S1{}
	t.register(6)
	do(mut t)
	println(t.i)
}
