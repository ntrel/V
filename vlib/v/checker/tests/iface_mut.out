vlib/v/checker/tests/iface_mut.vv:22:9: error: `S1` incorrectly implements method `register` of interface `Register`: receiver mutability differs
   20 |     mut t := S1{}
   21 |     t.register(6)
   22 |     do(mut t)
      |            ^
   23 |     println(t.i)
   24 | }
